packages/sync/src/offline/OfflineQueue.ts

```diff
@@ -23,79 +27,91 @@ export interface OfflineQueueOptions {
  * - persisting operation queue in provided storage
  * - updating client IDs with server IDs (explained below)
  */
-export class OfflineQueue extends OperationQueue {
+export class OfflineQueue {
+  public queue: OperationQueueEntry[] = [];
   private readonly storage?: PersistentStore<PersistedData>;
   private readonly storageKey?: string;
   private readonly listener?: OfflineQueueListener;
   private readonly state?: ObjectState;
 
-  constructor(options: OfflineQueueOptions) {
-    super(options);
-
-    const { storage, storageKey, listener, conflictStateProvider } = options;
+  constructor(options: OfflineLinkOptions) {
+    this.storage = options.storage;
+    this.storageKey = options.storageKey;
+    this.listener = options.listener;
+    this.state = options.conflictStateProvider;
+  }
 
-    this.storage = storage;
-    this.storageKey = storageKey;
-    this.listener = listener;
-    this.state = conflictStateProvider;
+  public enqueue(operation: Operation, forward: NextLink) {
+    const operationEntry = new OperationQueueEntry(operation, forward);
+    this.enqueueEntry(operationEntry);
+    return new Observable((observer) => {
+      operationEntry.observer = observer;
+      return () => ({ isOffline: true });
```

Comment:
isOffline: true is confusing as it is not used for anything at the moment. 
This function is getting called when client unsubscribes which in context of the links is going to be called only when client cancels request (like someone called destroy or something).
With that in mind I think we should not support canceling those for the moment and leave TODO like in http link: 
https://github.com/apollographql/apollo-link/blob/master/packages/apollo-link-http/src/httpLink.ts#L187-L191

@StephenCoady WDYT?

Reply:
```suggestion
      return () => { 
        // TODO support for canceling offline requests.
        return; 
      }
```